tests/test_resolve_names.cpp

```diff
@@ -0,0 +1,65 @@
+//   Copyright 2017 otris software AG
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+//
+//   This project is hosted at https://github.com/otris
+
+#include "fixtures.hpp"
+
+#include <cstring>
+#include <string>
+#include <vector>
+
+#ifdef EWS_USE_BOOST_LIBRARY
+namespace tests
+{
+TEST_F(ResolveNamesTest, NoUserFound)
+{
+    set_next_fake_response(read_file(assets_dir() / "resolve_names_error.xml"));
+
+    EXPECT_THROW(
+        {
+            auto result = service().resolve_names(
+                "name", ews::search_scope::active_directory);
+        },
+        ews::exception);
+}
```

Comment:
1. Good test.
2. You don't document that you are throwing in case nothing is found in the corresponding API docs.
3. That being said, we should seriously consider not throwing here even though it is an actual EWS error response. Throwing exceptions in C++ should not be done for normal control flow.